elephant/causality/granger.py

```diff
@@ -0,0 +1,438 @@
+# -*- coding: utf-8 -*-
```

Comment:
The causality submodule works a bit different than other modules of Elephant in that it's a proper submodule. In other cases, we had a top level python file that had top-level function calls, whereas `causality` this is more a grouping submodule like `random.random` in numpy. I think it's worth discussing if we want this or if we want to stick to a flat hierarchy (which depends also on what elese we envision to be captured by `causality`. In the former case, somehow the mapping of the module to the documentation is not quite clear to me -- currently it seems to me the generic `causality` and the specific `granger` is still a bit mixed up.

Reply:
We can definitely discuss the pros and cons of this. Also, we could have a functional connectivity module encompassing corr, mutual info, cross-corr, Granger in the time-domain and coherence, phase methods, Granger in the Frequency domain as per [this figure](https://www.frontiersin.org/files/Articles/165147/fnsys-09-00175-HTML/image_m/fnsys-09-00175-g001.jpg). Full paper [here](https://www.frontiersin.org/articles/10.3389/fnsys.2015.00175/full).